### PR TITLE
Augment RegisterResponderWithQuery function

### PR DESCRIPTION
**Pull request by maxatome**
- allow to pass several values for one query parameter;
- fix the case when RegisterResponder register an url with unsorted
  query params: RoundTrip() now checks with url & untouched query
  params before falling back to sorting query params;
- code simplified to rely on net/url.Values.

Could replace #55 & #57 what do you think of this, @Zanadar and @kardolus?

`RegisterResponderWithQueryValues` can be renamed if you have a better idea.

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -196,40 +215,64 @@ func (m *MockTransport) RegisterResponder(method, url string, responder Responde
...
-	url := path
-	mapQuery := make(map[string][]string, len(query))
+	mapQuery := make(url.Values, len(query))
 	for key, e := range query {
 		mapQuery[key] = []string{e}
 	}
-	queryString := mapToSortedQuery(mapQuery)
-	if queryString != nil {
-		url = path + "?" + *queryString
+	if queryString := sortedQuery(mapQuery); queryString != "" {
```
Comment (Zanadar) [inline transport.go]: At this point you could do `return RegisterResponderWithQueryValues(method, mapQuery, responder)`

File: transport_test.go [hunk 2]
```diff
@@ -208,21 +208,53 @@ func TestMockTransportPathOnlyFallback(t *testing.T) {
 	defer DeactivateAndReset()
 
 	for responder, paths := range map[string][]string{
-		"/hello/world?query=string#fragment": {
-			testUrl + "hello/world?query=string#fragment",
+		// unsorted query string matches exactly
```
Comment (Zanadar) [inline transport_test.go]: Nice test additions

File: transport.go (hunk 1 above)
Comment (maxatome) [reply transport.go]: done

**Comment by maxatome** [comment]
I agree that having both `RegisterResponderWithQueryValues` and `RegisterResponderWithQuery` is confusing. I agree too that `RegisterResponderWithQuery` should never be released as is, but it is done now. Refering to `RegisterResponderWithQueryValues` in `RegisterResponderWithQuery` is a step to avoid confusion...

Of course we can bump the major version, but I prefer not. IMO having a new branch will add more confusion than having 2 similar but well documented methods, even if the API is not as clean as it could be in this case.

Note that if you have a better name for `RegisterResponderWithQueryValues`, do not hesitate to propose it.

By the way, `RegisterResponderWithQuery` now calls `RegisterResponderWithQueryValues` as you rightly suggested.

**Comment by maxatome** [comment]
@Zanadar I just merged `RegisterResponderWithQueryValues` in `RegisterResponderWithQuery`. If this design seems correct to you, I will complete unit testing of this stuff.

**Comment by maxatome** [comment]
Unit tests OK.

**Pull request merged**